### PR TITLE
[Keyboard] Idobao ID80 V3

### PR DESCRIPTION
**Pull request by peepeetee**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/idobao/id80/v3/config.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+/*== or choose animations ==*/
+//    #define RGBLIGHT_EFFECT_BREATHING
+//    #define RGBLIGHT_EFFECT_RAINBOW_MOOD
+//    #define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+//    #define RGBLIGHT_EFFECT_SNAKE
+//    #define RGBLIGHT_EFFECT_KNIGHT
+//    #define RGBLIGHT_EFFECT_CHRISTMAS
+//    #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//    #define RGBLIGHT_EFFECT_RGB_TEST
+//    #define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline keyboards/idobao/id80/v3/config.h:85]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/idobao/id80/v3/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    if (index == 0) {
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+    return true;
+}
+
```
Comment (drashna) [inline keyboards/idobao/id80/v3/keymaps/default/keymap.c:47]:
```suggestion
```

File: keyboards/idobao/id80/v3/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) {
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/idobao/id80/v3/keymaps/via/keymap.c:62]:
```suggestion
```

File: keyboards/idobao/id80/v3/v3.c [hunk 4]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2022 peepeetee (@peepeetee)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "v3.h"
```
Comment (drashna) [inline keyboards/idobao/id80/v3/v3.c:4]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "v3.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**